qgrid/grid.py

```diff
@@ -214,16 +216,26 @@ class QGridWidget(widgets.DOMWidget):
     df = Instance(pd.DataFrame)
     precision = Integer(6)
     grid_options = Unicode('', sync=True)
-    remote_js = Bool(True)
+    remote_js = Bool(False)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize all variables before building the table."""
+        kwargs.setdefault('grid_options', json.dumps(defaults.grid_options))
```

Comment:
That API remains. The new release of traitlets has a *new* API that is much
nicer, but the old API still works fine (it is deprecated). For now I would
stick with the older API and everything should be fine on both Jupyter 3
and 4 (from a traitlets perspective).

On Sun, Dec 13, 2015 at 1:25 PM, Scott Sanderson <notifications@github.com>
wrote:

> In qgrid/grid.py
> <https://github.com/quantopian/qgrid/pull/49#discussion_r47450399>:
>
> > @@ -214,16 +216,26 @@ class QGridWidget(widgets.DOMWidget):
> >      df = Instance(pd.DataFrame)
> >      precision = Integer(6)
> >      grid_options = Unicode('', sync=True)
> > -    remote_js = Bool(True)
> > +    remote_js = Bool(False)
> > +
> > +    def __init__(self, *args, **kwargs):
> > +        """Initialize all variables before building the table."""
> > +        kwargs.setdefault('grid_options', json.dumps(defaults.grid_options))
>
> cc @ellisonbg <https://github.com/ellisonbg>. Do you know of any projects
> that are using traitlets in an IPython 3/4 compatible way? In the old
> traitlets, we would use _grid_options_default here, but I think that API
> is deprecated and/or removed in the latest traitlets release?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/quantopian/qgrid/pull/49/files#r47450399>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com